Backend/Controllers/AuthenticationController.cs

```diff
@@ -79,4 +80,35 @@ public IQueryable<Session> GetSessions([FromAuthentication] User user)
     [HttpGet]
     public User GetAuthUser([FromAuthentication] User user) => user;
 
+    [HttpPost("reset/start")]
+    public async Task InitPasswordReset(string email) {
+        email = email.ToLowerInvariant();
+        var user = await _context.Users.FirstOrDefaultAsync(x => x.Email == email);
+        if (user == null) { return; }
+
+        var token = await _context.ResetTokens.AddAsync(new() {
+            User = user,
+            Token = RandomNumberGenerator.GetBytes(64),
+        });
+        await _context.SaveChangesAsync();
+
+        var textToken = Base64UrlTextEncoder.Encode(token.Entity.Token);
+        // TODO: Send via email
+    }
+    
+    [HttpPost("reset/finish")]
+    public async Task<IActionResult> PasswordReset(string textToken, string newPassword) {
```

Comment:
Even though this is not much, since in FE we are most likely forced to use a model class to get validation, I'd use the model here as well.

Reply:
I disagree, there's no reason to take in redundant data (e.g. confirm password) just to make it fit the internals of the frontend. Simply do the necessary conversions there.

Reply:
You don't take in confirm password data since it's not in the json

Reply:
If I'm not mistaken you should be able to just send the model anyways and it automatically deconstructs the model into the parameters?

Reply:
Completely forgot that you cannot even take 2 parameters from a requestbody like this

Reply:
So you dont have to use the model in the backend, FE just sends Token and Password as json

Reply:
The above two statements appear contradictory??

Reply:
Its the same thing as with the tags yesterday. You need to annotate the parameters with [FromBody] because string is a simple type but you cannot annotate two parameters.

Reply:
You can however map the body to something without explicity using a model class

Reply:
yes i think thats what i saw